static/app/views/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -0,0 +1,323 @@
+import {Component} from 'react';
+import {createFilter} from 'react-select';
+import debounce from 'lodash/debounce';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import {SentryAppInstallation} from 'app/types';
+import {replaceAtArrayIndex} from 'app/utils/replaceAtArrayIndex';
+import withApi from 'app/utils/withApi';
+import FieldFromConfig from 'app/views/settings/components/forms/fieldFromConfig';
+import Form from 'app/views/settings/components/forms/form';
+import FormModel from 'app/views/settings/components/forms/model';
+import {Field, FieldValue} from 'app/views/settings/components/forms/type';
+
+// 0 is a valid choice but empty string, undefined, and null are not
+const hasValue = value => !!value || value === 0;
+
+export type FieldFromSchema = Omit<Field, 'choices' | 'type'> & {
+  type: 'select' | 'textarea' | 'text';
+  default?: 'issue.title' | 'issue.description';
+  uri?: string;
+  depends_on?: string[];
+  choices?: Array<[any, string]>;
+  async?: boolean;
+};
+
+export type Config = {
+  uri: string;
+  required_fields?: FieldFromSchema[];
+  optional_fields?: FieldFromSchema[];
+};
+
+// only need required_fields and optional_fields
+type State = Omit<Config, 'uri'> & {
+  optionsByField: Map<string, Array<{label: string; value: any}>>;
+};
+
+type Props = {
+  api: Client;
+  sentryAppInstallation: SentryAppInstallation;
+  appName: string;
+  config: Config;
+  action: 'create' | 'link';
+  element: 'issue-link' | 'alert-rule-action';
+  extraFields?: {[key: string]: any};
+  extraRequestBody?: {[key: string]: any};
+  onSubmitSuccess: (...params: any[]) => void;
+  getFieldDefault?: (field: FieldFromSchema) => string;
+};
+
+export class SentryAppExternalForm extends Component<Props, State> {
+  state: State = {optionsByField: new Map()};
+
+  componentDidMount() {
+    this.resetStateFromProps();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.action !== this.props.action) {
+      this.model.reset();
+      this.resetStateFromProps();
+    }
+  }
+
+  model = new FormModel();
+
+  // reset the state when we mount or the action changes
+  resetStateFromProps() {
+    const {config, action, extraFields} = this.props;
+    this.setState({
+      required_fields: config.required_fields,
+      optional_fields: config.optional_fields,
+    });
+    // we need to pass these fields in the API so just set them as values so we don't need hidden form fields
+    this.model.setInitialData({
+      ...extraFields,
```

Comment:
As stated above, the Issue Linking API requires an initial value of `groupId`.  This allow the parent component specify this.